src/Microsoft.AspNetCore.Mvc.Testing/WebApplicationFactory.cs

```diff
@@ -0,0 +1,313 @@
+// Copyright (c) .NET  Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Reflection;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyModel;
+
+namespace Microsoft.AspNetCore.Mvc.Testing
+{
+    /// <summary>
+    /// Factory for bootstrapping an application in memory for functional end to end tests.
+    /// </summary>
+    /// <typeparam name="TEntryPoint">A type in the entry point assembly of the application.
+    /// Typically the Startup or Program classes can be used.</typeparam>
+    public class WebApplicationFactory<TEntryPoint> : IDisposable where TEntryPoint : class
+    {
+        private TestServer _server;
+        private IWebHostBuilder _builder;
+        private IList<HttpClient> _clients = new List<HttpClient>();
+
+        /// <summary>
+        /// <para>
+        /// Creates a TestServer instance using the MVC application defined by <typeparamref name="TEntryPoint"/>.
```

Comment:
This **constructor** doesn't create a `TestServer` instance (in fact, it does nothing at all), so this comment is a bit misleading.